Source/CParse/templ.c

```diff
@@ -371,11 +370,20 @@ int Swig_cparse_template_expand(Node *n, String *rname, ParmList *tparms, Symtab
 	  sz = Len(typelist);
 	  for (i = 0; i < sz; i++) {
 	    String *s = Getitem(typelist, i);
-	    /*      Replace(s,name,value, DOH_REPLACE_ID); */
-	    /*      Printf(stdout,"name = '%s', value = '%s', tbase = '%s', iname='%s' s = '%s' --> ", name, dvalue, tbase, iname, s); */
-	    SwigType_typename_replace(s, name, dvalue);
-	    SwigType_typename_replace(s, tbase, iname);
-	    /*      Printf(stdout,"'%s'\n", s); */
+	    /*
+	      The approach of 'trivially' replacing template arguments is kind of fragile.
+	      In particular if types with similar name in different namespaces appear.
+	      We will not replace template args if a type/class exists with the same
+	      name which is not a template.
+	    */
+	    Node * __n = Swig_symbol_clookup(s, 0);
+	    String * __nn = __n ? Getattr(__n, "sym:name") : 0;
+	    String * nn = __nn ? Getattr(n, "sym:name") : 0;
+	    int __eq = __nn && nn ? Strcmp(nn, __nn) : 1;
```

Comment:
This is confusing code.  The variable is called `__eq` which suggests it's true if equal, but (when both the arguments are non-NULL) it's set to the result of `Strcmp()`, which is `0` if equal and non-zero if non-equal.  So if the strings are equal, it seems `__eq` is false!?!

Better to compare the strings with `Equal(a, b)` and make the sense of the variable match what its name suggests.

Reply:
done